airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java

```diff
@@ -104,9 +104,9 @@
                 .map(s -> s.getAlgorithm()).collect(Collectors.toList());
 
         if (algorithms.contains(algorithmNative)) {
-            random = secureRandom.apply(algorithmNative);
+            random = createSecureRandom(algorithmNative);
         } else if (algorithms.contains(algorithmSHA)) {
-            random = secureRandom.apply(algorithmSHA);
+            random = createSecureRandom(algorithmSHA);
```

Comment:
This still has multiple uses of the `algorithmXXX` values, so I still think my prior comment suggesting iterating over the list of algorithms to use and applying the first one that matches would make this cleaner.

Reply:
How about doing something like this?

```diff
diff --git a/airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java b/airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java
index ebc4dba9..abd74f5b 100644
--- a/airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java
@@ -96,17 +96,14 @@ public class SearchServiceUtilities {
 
     {
 
-        String algorithmNative = "NativePRNG";
-        String algorithmSHA = "SHA1PRNG";
-
         List<String> algorithms = Arrays.asList(Security.getProviders()).stream()
                 .flatMap(p -> p.getServices().stream().filter(s -> "SecureRandom".equals(s.getType())))
                 .map(s -> s.getAlgorithm()).collect(Collectors.toList());
 
-        if (algorithms.contains(algorithmNative)) {
-            random = createSecureRandom(algorithmNative);
-        } else if (algorithms.contains(algorithmSHA)) {
-            random = createSecureRandom(algorithmSHA);
+        for (String algorithmName : new String[]{"NativePRNG", "SHA1PRNG"}) {
+            if (algorithms.contains(algorithmName)) {
+                random = createSecureRandom(algorithmName);
+            }
         }
 
         if (isEmpty(random)) {
```

Reply:
Yes, something like this is what I had in mind.  Seems a lot cleaner.  I'd like to see this cleanup done and get this merged.